src/config/mod.rs

```diff
@@ -537,12 +538,17 @@ macro_rules! build_cf_opt {
         $cf_name:ident,
         $cache:expr,
         $compaction_limiter:expr,
-        $region_info_provider:ident
+        $region_info_provider:ident,
+        $engine_type:ident,
     ) => {{
         let mut block_base_opts = BlockBasedOptions::new();
         block_base_opts.set_block_size($opt.block_size.0 as usize);
         block_base_opts.set_no_block_cache($opt.disable_block_cache);
         block_base_opts.set_block_cache($cache);
+        if $engine_type == EngineType::RaftKv2 {
+            block_base_opts.set_partition_filters(true);
```

Comment:
Maybe these should be part of (private) configuration and be changed in `optimize_for`.

Reply:
OK. I think we can make it public configuration as it's quite useful for both multi-rocksdb and single-rocksdb. And meanwhile, the benchmark shows that it does have noticeable regression in normal scenarios.  

<img width="1805" alt="Screen Shot 2023-03-30 at 9 52 02 PM" src="https://user-images.githubusercontent.com/9442422/229025787-69f8e85d-9db4-48b3-ad77-06d75cb57a4d.png">